src/Orleans/Core/GrainExtensions.cs

```diff
@@ -211,5 +215,11 @@ private static void ThrowIfNullGrain(IAddressable grain)
                 throw new ArgumentNullException(nameof(grain));
             }
         }
+
+        public static StreamIdentity GetImplicitStreamIdentity(this IGrainWithGuidCompoundKey grain)
+        {
```

Comment:
GrainExtention is not a good place to put this method, since this method is not commonly applicable to all grains. This method only applies to grains which subscribes to a stream implicitly.